noxfile.py

```diff
@@ -0,0 +1,143 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Final, Literal, TypeAlias
+
+import nox
+
+# -- Parameter sets --
+ModelSubpackagePath: TypeAlias = Literal[
+    "atmosphere/advection",
+    "atmosphere/diffusion",
+    "atmosphere/dycore",
+    "atmosphere/subgrid_scale_physics/microphysics",
+    "common",
+    "driver",
+    # "testing", #TODO: Add tests to testing subpackage
+]
+ModelTestsSubset: TypeAlias = Literal["datatest", "stencils", "basic"]
+
+MODEL_SUBPACKAGE_PATHS: Final[Sequence[str]] = ModelSubpackagePath.__args__
+MODEL_TESTS_SUBSETS: Final[Sequence[str]] = ModelTestsSubset.__args__
+
+# -- nox configuration --
+nox.options.default_venv_backend = "uv"
+nox.options.sessions = ["test_model", "test_tools"]
+
+
+# -- nox sessions --
+# Model benchmark sessions
+# TODO(egparedes): Add backend parameter
+# TODO(edopao,egparedes): Change 'extras' back to 'all' once mpi4py can be compiled with hpc_sdk
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
+    """Run pytest benchmarks for selected icon4py model subpackages."""
+    _install_session_venv(session, extras=["dace", "fortran", "io", "testing"], groups=["test"])
+
+    with session.chdir(f"model/{subpackage}"):
+        session.run(*"pytest -sv --benchmark-only".split(), *session.posargs)
+
+
+# Model test sessions
+# TODO(egparedes): Add backend parameter
+# TODO(edopao,egparedes): Change 'extras' back to 'all' once mpi4py can be compiled with hpc_sdk
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+@nox.parametrize("selection", MODEL_TESTS_SUBSETS)
+def test_model(session: nox.Session, selection: ModelTestsSubset, subpackage: ModelSubpackagePath) -> None:
+    """Run tests for selected icon4py model subpackages."""
+    _install_session_venv(session, extras=["dace", "fortran", "io", "testing"], groups=["test"])
```

Comment:
Suggestion for cleanup of my change:
```suggestion
    _install_session_venv(session, extras=["dace", "io", "testing"], groups=["test"])
```